docs/core/diagnostics/trace-perfcollect-lttng.md

```diff
@@ -0,0 +1,241 @@
+---
+title: Tracing .NET applications with PerfCollect.
+description: A tutorial that walks you through collecting a trace with perfcollect in .NET.
+ms.topic: tutorial
+ms.date: 10/23/2020
+---
+
+# Tracing .NET applications with PerfCollect
+
+**This article applies to: ✔️** .NET Core 2.1 SDK and later versions
+
+When performance problems are encountered on Linux, collecting a trace with `perfcollect` can be used to gather detailed information about what was happening on the machine at the time of the performance problem.
+
+`perfcollect` is a bash script that leverages [Linux Tracing Tookit-Next Generation (LTTng)](https://lttng.org) to collect events written from the runtime or any [EventSource](xref:System.Diagnostics.Tracing.EventListener), as well as [perf](https://perf.wiki.kernel.org/) to collect CPU samples of the target process.
+
+## Preparing Your Machine
+
+Follow these steps to prepare your machine to collect a performance trace with `perfcollect`.
+
+> [!NOTE]
+> If you are in a container environment, your container needs to have `SYS_ADMIN` capability. For more information on tracing applications inside container using PerfCollect, refer to [Collect diagnostics in containers](./diagnostics-in-containers.md) documentation.
+
+1. Download `perfcollect`.
+
+    > ```bash
+    > curl -OL http://aka.ms/perfcollect
+    > ```
+
+2. Make the script executable.
+
+    > ```bash
+    > chmod +x perfcollect
+    > ```
+
+3. Install tracing prerequisites - these are the actual tracing libraries.
+
+    > ```bash
+    > sudo ./perfcollect install
+    > ```
+
+    This will install the following prerequisites on your machine:
+
+    1. `perf`: the Linux Performance Events sub-system and companion user-mode collection/viewer application. `perf` is part of the Linux kernel source, but is not usually installed by default.
+
+    2. `LTTng`: Used to capture event data emitted at runtime by CoreCLR. This data is then used to analyze the behavior of various runtime components such as the GC, JIT and thread pool.
+
+## Collecting a Trace
+
+1. Have two shells available - one for controlling tracing, referred to as **[Trace]**, and one for running the application, referred to as **[App]**.
+
+2. **[Trace]** Start collection.
+
```

Comment:
Not sure if you want to include it here, but there is also some support for threadtime traces as well.  They really only view well on PerfView, and I consider them experimental, just because I haven't fully proved them out.  They are collectible by specifying `-threadtime`.

Reply:
Thanks Brian. In that case, I think I'll leave out the `-threadtime` flag until we are more confident about using this. I will be updating the docs frequently in the future so maybe this is something we can add in the next iteration for this doc along with what Christophe is adding via his PR.

Reply:
Sounds good.